### PR TITLE
MINOR HT-37 - Add functionality for requesting a pickup

### PR DESCRIPTION
**Pull request by stewartbebbington**

### CODE REVIEW

File: HaulThis.Test/ViewModels/PickupRequestModelTests.cs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        Status = "Pending"
+      };
+      _mockPickupRequestService
+          .Setup(service => service.GetPickupRequestInfo(It.IsAny<int>()))
+          .ReturnsAsync(expectedRequest);
+
+      _viewModel.RequestId = 1;
+
+      // Act
+      _viewModel.GetPickupRequestInfoCommand.Execute(null);
```
Comment (conor27nash) [inline HaulThis.Test/ViewModels/PickupRequestModelTests.cs:38]: This method called is asynchronous should we have an await before it to make sure it completes properly. This is the same for the other two tests

File: HaulThis/Services/PickupRequestService.cs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+      using var loggerFactory = LoggerFactory.Create(builder =>
+      {
+        builder.AddConsole();
+        builder.AddDebug();
+      });
+      _logger = loggerFactory.CreateLogger<PickupRequestService>();
+    }
+
+    public async Task<PickupDeliveryRequest?> GetPickupRequestInfo(int id)
```
Comment (jamieyoung5) [inline HaulThis/Services/PickupRequestService.cs:39]: we can get pickup requests but cannot make them?

File: HaulThis/ViewModels/PickupRequestModel.cs
Comment (jamieyoung5) [inline HaulThis/ViewModels/PickupRequestModel.cs:1]: to correspond to the mvvm architecture, this data model should reside in Models/ and this should be like the ui's link to our 'backend' (the services etc) to map data to views or get data for views

File: HaulThis/Services/PickupRequestService.cs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+            FROM 
+                PickupDeliveryRequests
+            WHERE 
+                Id = @p0";
+
+    public PickupRequestService(IDatabaseService databaseService)
+    {
+      _databaseService = databaseService ?? throw new ArgumentNullException(nameof(databaseService));
+
+      using var loggerFactory = LoggerFactory.Create(builder =>
```
[the remark this answers is not in the document]
Comment (stewartbebbington) [reply HaulThis/Services/PickupRequestService.cs:31]: done

File: HaulThis/Services/PickupRequestService.cs (hunk 2 above)
Comment (stewartbebbington) [reply HaulThis/Services/PickupRequestService.cs:40]: done

File: HaulThis/ViewModels/PickupRequestModel.cs
Comment (stewartbebbington) [reply HaulThis/ViewModels/PickupRequestModel.cs:1]: done

File: HaulThis.Test/Services/PickupRequestServiceTests.cs [hunk 4]
```diff
@@ -0,0 +1,139 @@
+using System;
+using System.Data;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using HaulThis.Models;
+using HaulThis.Services;
+
```
Comment (conor27nash) [inline HaulThis.Test/Services/PickupRequestServiceTests.cs:7]: Missing namespace

File: HaulThis.Test/ViewModels/PickupRequestModelTests.cs [hunk 5]
```diff
@@ -0,0 +1,130 @@
+using HaulThis.Models;
+using HaulThis.Services;
+
```
Comment (conor27nash) [inline HaulThis.Test/ViewModels/PickupRequestModelTests.cs:3]: missing namespace

**Review by jamieyoung5 — APPROVED** [review]

**Pull request merged**